site/content/features/switch.md

```diff
@@ -0,0 +1,319 @@
+
+The following table shows the four forms along the two axes (expression/statement and disjoint branches/fall through).
+
+{{< rawhtml >}}
+<table>
+<tr>
+<th>  </th> <th> Expression </th> <th> Statement </th>
+</tr>
+<tr></tr>
+<tr>
+<th> No fall through </th>
+<td>
+<pre>
+int numLetters = switch (seasonName) {
+   case "Summer", "Winter" -> 6;
+   case "Spring" -> {
+      System.out.println("spring time!");
+      yield 6;
+   }
+   case "Fall" -> 4;
+   default -> -1;
+};
+```
+</pre>
+</td>
+<td>
+<pre>
+switch (seasonName) {
+   case "Summer", "Winter" ->
+      numLetters = 6;
+   case "Spring" -> {
+      System.out.println("spring time!");
+      numLetters = 4;
+   }
+   case "Fall" -> 4;
+      numLetters = 4;
+   default ->
+      numLetters = -1;
+}
+</pre>
+</td>
+</tr>
+<tr></tr>
+<tr>
+<th> Fall through </th>
+<td>
+<pre>
+int numLetters = switch (seasonName) {
+   case "Spring":
+      System.out.println("spring time!");
+   case "Summer", "Winter":
+      yield 6;
+   case "Fall":
+      yield 4;
+   default:
+      yield -1;
+};
+</pre>
+</td>
+<td>
+<pre>
+switch (seasonName) {
+   case "Spring":
+      System.out.println("spring time!");
+   case "Summer", "Winter":
+      numLetters = 6;
+      break;
+   case "Fall":
+      numLetters = 4;
+      break;
+   default:
+      numLetters = -1;
+}
+</pre>
+</td>
+</tr>
+</table>
+{{< /rawhtml >}}
+
+## Recommendations
+
+Prefer `switch` expressions over `switch` statements. If  a `switch` or `if/else if/else` statement sets a variable in each branch, use a single assignment with a `switch` expression instead. 
+
+```
+int numLetters = switch (seasonName) {
+    case "Spring", "Summer", "Winter" -> 6
+    case "Fall" -> 4
+    default -> -1
+};
+```
+
+is better than
+
+```
+switch (seasonName) {
+    case "Spring", "Summer", "Winter" -> 
+        numLetters = 6;
+    case "Fall" -> 
+        numLetters = 4;
+    default -> 
+        numLetters = -1;
+}
+```
+
+Minimize `break` and `yield`. Use them only if you actually need the "fall through" or "statement before expression" behavior. This is very uncommon. 
+
+The `yield` statement may in time join the labeled `break/continue` statements: unloved by all except the authors of certification exam questions.
+
+## Summary
+
+1. There are four forms of `switch`: statements and expressions, with and without fall through
+2. A `switch` expression selects one or more values, matching an operand. 
+3. The cases in a `switch` expression must be disjoint and exhaustive
+4. A `switch` expression or statemnt with `->` tokens does not have fall through behavior
+5. A `case` label can be followed by multiple constants, separated by commas
+6. In a `switch` expression, the `yield` statement exits and yields a value
+7. In a `switch` expression without fall through, the `->` token can be followed by a block statement that yields a value
+
+
```

Comment:
Should we add a references section as with the other articles?
```suggestion
## References

* [JEP 325: Switch Expressions (Preview), OpenJDK](https://openjdk.java.net/jeps/325)
* [JEP 354: Switch Expressions (Second Preview), OpenJDK](https://openjdk.java.net/jeps/354)
* [JEP 361: Switch Expressions, OpenJDK](https://openjdk.java.net/jeps/361)
```

Reply:
Yes, we should. I can't commit this from here--it's "outdated". I pushed again with the reference section.